stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -1433,12 +1433,12 @@ function(add_swift_target_library_single target name)
     ENABLE_ASSERTIONS "${enable_assertions}"
     ANALYZE_CODE_COVERAGE "${analyze_code_coverage}"
     ENABLE_LTO "${lto_type}"
-    DEPLOYMENT_VERSION_OSX "${SWIFTLIB_DEPLOYMENT_VERSION_OSX}"
-    DEPLOYMENT_VERSION_MACCATALYST "${SWIFTLIB_DEPLOYMENT_VERSION_MACCATALYST}"
-    DEPLOYMENT_VERSION_IOS "${SWIFTLIB_DEPLOYMENT_VERSION_IOS}"
-    DEPLOYMENT_VERSION_TVOS "${SWIFTLIB_DEPLOYMENT_VERSION_TVOS}"
-    DEPLOYMENT_VERSION_WATCHOS "${SWIFTLIB_DEPLOYMENT_VERSION_WATCHOS}"
-    DEPLOYMENT_VERSION_XROS "${SWIFTLIB_DEPLOYMENT_VERSION_XROS}"
+    DEPLOYMENT_VERSION_OSX "${SWIFTLIB_SINGLE_DEPLOYMENT_VERSION_OSX}"
```

Comment:
Nice catch, thanks for taking care of this!

Turns out this has been working by accident so far since [in CMake variables defined by a function are visible in any other function called by it](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#variables) -- since in general we tend to call `add_swift_target_library`, when we get here we are able to see `SWIFTLIB_DEPLOYMENT_VERSION_*` that the caller defined.

I don't foresee issues in the scenarios in which we are calling `add_swift_target_library_single` directly (embedded, and building part of the stdlib to support bootstrapping) -- in those we are not forcing a specific deployment target (if we were, we would have noticed this issue before).